File: backend/app/schemas/favorite.py
from datetime import datetime

from pydantic import BaseModel


class FavoriteRead(BaseModel):
    id: str
    job_post_id: str
    job_title: str
    company_name: str
    location_code: str | None = None
    close_at: str | None = None
    created_at: datetime


class FavoriteListResponse(BaseModel):
    items: list[FavoriteRead]
    total: int

File: frontend/src/types/favorite.ts
export interface FavoriteRead {
  id: string;
  job_post_id: string;
  job_title: string;
  company_name: string;
  location_code: string | null;
  close_at: string | null;
  created_at: string;
}

export interface FavoriteListResponse {
  items: FavoriteRead[];
  total: number;
}
